sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -154,6 +155,24 @@ private static String encode(final String stringToEncode) {
         }
     }
 
+    /**
+     * Performs a safe encoding of a url string, only encoding the path.
+     *
+     * @param url The url to encode.
+     * @return The encoded url.
+     */
+    public static String encodeUrlPath(String url) {
+        /* Deconstruct the URL and reconstruct it making sure the path is encoded. */
+        UrlBuilder builder = UrlBuilder.parse(url);
+        String path = builder.getPath();
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+        path = Utility.urlEncode(Utility.urlDecode(path));
```

Comment:
do we have to decode it first ? Wouldn't already encoded chars be left alone during encoding?

Reply:
We need to do this to preserve the case where a user has a URL that looks encoded. - Like if they pass in a url with hello%20hello - if we encode that it'll become hello(whatever % encoding is)20hello, which we don't want to do

Reply:
That makes sense.